modules/serverless-functions-using-integrated-registry.adoc

```diff
@@ -0,0 +1,24 @@
+// Module included in the following assemblies:
+//
+// * serverless/serverless-functions-setup.adoc
+
+[id="serverless-functions-using-integrated-registry_{context}"]
+= Building and deploying functions with OpenShift Container Registry
+
+When building and deploying functions, the resulting container image is stored in an image registry. Usually this will be a public registry, such as Quay. However, you can use the integrated OpenShift Container Registry instead if it has been exposed by a cluster administrator.
+
+.Procedure
+
+* Run the `kn func build` or `kn func deploy` commands with the OpenShift Container Registry specified for the `-r` parameter:
++
+.Example build command
+[source,terminal]
```

Comment:
My concern is here on both build and deploy is that -i option will take precedence over -r. So in case user pass both -i (without openshift registr) and -r (with openshift registry), func will not use it.

Also, I might be wrong but, the url below (as is `image-registry.openshift-image-registry.svc`)  may not work. I think user/developer will need to replace that with the actual URL exposed by the cluster  ( the one returned from `oc get route -n openshift-image-registry`)

Reply:
So would you remove the `-i` option?
Re the URL, can you confirm whether we need to add `oc get route -n openshift-image-registry` here as a first step maybe @lance if users do indeed need the actual URL?

Reply:
I agree with @jrangelramos that the `-i` flag shouldn't be necessary here and may confuse the matter. Likewise, I think there's no need to specify the `-n` flag here either, as it is not relevant to what is being illustrated here. And yes, the URL exposed by the cluster. This could be done as a single command like so:

```
$ kn func deploy -r $(oc get route -n openshift-image-registry)
```

Or with two commands, you could set an environment variable for use in the next command.

```
$ export REGISTRY=$(oc get route -n openshift-image-registry)
$ kn func deploy -r $REGISTRY
```

Or, finally, you could set the `FUNC_REGISTRY` environment variable, and have it picked up automagically.

```
$ export FUNC_REGISTRY=$(oc get route -n openshift-image-registry)
$ kn func deploy
```

But since we haven't really documented these env vars, probably want to avoid that.

Reply:
I think it makes most sense to go with `$ kn func deploy -r $(oc get route -n openshift-image-registry)` since it's the simplest way; updating in next commit.